libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts

```diff
@@ -104,29 +111,44 @@ describe('Selection box grid component', () => {
   });
 
   it(`should update CSS responsive classes on window resize`, async(() => {
-    const spy = spyOn(
-      component.selectionBoxGrid as any,
-      'updateBreakpointClass'
+    const setResponsiveClassSpy = spyOn(
+      SkySelectionBoxAdapterService.prototype,
+      'setResponsiveClass'
     );
-    expect(spy).not.toHaveBeenCalled();
+    spyOn(
+      SkySelectionBoxAdapterService.prototype,
+      'getParentWidth'
+    ).and.returnValue(300);
+    expect(setResponsiveClassSpy).not.toHaveBeenCalled();
 
     SkyAppTestUtility.fireDomEvent(window, 'resize');
     fixture.detectChanges();
 
-    expect(spy).toHaveBeenCalledTimes(1);
+    expect(setResponsiveClassSpy).toHaveBeenCalledTimes(1);
   }));
 
-  it(`should recalculate heights when child DOM elements change`, async(() => {
-    const spy = spyOn(
-      component.selectionBoxGrid as any,
-      'updateChildrenHeights'
+  it(`should recalculate heights when child DOM elements change`, waitForAsync(async () => {
+    const resetHeightSpy = spyOn(
+      SkyCoreAdapterService.prototype,
+      'resetHeight'
+    );
+    const syncMaxHeightSpy = spyOn(
+      SkyCoreAdapterService.prototype,
+      'syncMaxHeight'
     );
-    expect(spy).not.toHaveBeenCalled();
+    spyOn(
+      SkySelectionBoxAdapterService.prototype,
+      'getParentWidth'
+    ).and.returnValue(300);
+    expect(resetHeightSpy).not.toHaveBeenCalled();
+    expect(syncMaxHeightSpy).not.toHaveBeenCalled();
 
     component.dynamicDescription = `Something really really really really really really really really really really really really really really really long to force a large height than before!`;
     fixture.detectChanges();
-    fixture.whenStable().then(() => {
-      expect(spy).toHaveBeenCalledTimes(1);
+    await fixture.whenStable();
+    setTimeout(() => {
```

Comment:
`setTimeout`? Should this `it` block use `fakeAsync`?

Reply:
So I actually met with Erika about this - happy to get on a call with you if needed. This test was suffering from an issue where the mutation observer isn't firing in time and the test is failing. We tried a ton of combinations of `fakeAsync` and `tick` calls and standard async with `whenStable` and/or `whenRenderingDone` calls and the only thing that we could get to work consistently was adding the `setTimeout`. Happy to chat with you though.

Reply:
We talked. Made a change.